client/implementation/op_snapshots_restore.go

```diff
@@ -25,8 +27,20 @@ func (c *defaultYBCliClient) SnapshotsRestore(opConfig *configs.OpSnapshotRestor
 	payload := &ybApi.RestoreSnapshotRequestPB{
 		SnapshotId: protoSnapshotID,
 	}
-	if opConfig.RestoreHt > 0 {
-		payload.RestoreHt = &opConfig.RestoreHt
+
+	if opConfig.RestoreAt > 0 {
+		payload.RestoreHt = &opConfig.RestoreAt
+	}
+	if opConfig.RestoreRelative > 0 {
+		serverClock, err := c.ServerClock()
+		if err != nil {
+			return nil, err
+		}
+		if serverClock.HybridTime == nil {
+			return nil, fmt.Errorf("no hybrid time in server clock response")
+		}
+		newHybridTime := *serverClock.HybridTime - utils.ClockTimestampToHTTimestamp(uint64(opConfig.RestoreRelative.Microseconds()))
```

Comment:
I'd love to understand why I need this `utils.ClockTimestampToHTTimestamp` call and why only on the relative value. My guess is that I get a hybrid clock from the server and my diff is just a regular _timestamp_ so I need to convert it into another hybrid clock and only then subtract.

Well, it does seem to pick the correct snapshot every time.